### PR TITLE
Setup branch

### PR DESCRIPTION
**Pull request by soesandarwin2201**
## What does this PR do?
This PR is raised to add the [devise](https://github.com/heartcombo/devise) gem to your app and use it for the registration and login of users.

## Description of the task to be completed
- [x] Create a new Rails app called 'hello-rails-react'.
- [x] Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- [x] Verify that the jsbundling-rails gem is added to your Gemfile.
- [x] Initialize your project with Git.
- [x] Install needed node dependencies as described in the previous lesson's tutorial.
- [x] Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- [x] Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- [x] Make sure that your project has a Postgres database set up.
- [x] Create a table for storing your messages and create 5 different greetings.
- [x] Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- [x] Create a static view that will be the root of your app.
- [x] Create the App component with react-router and render it in your static view.
- [x] Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- [x] Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- [x] Display the random greeting in your Greeting component.

## How should it be tested
Clone the repository
```
git clone https://github.com/soesandarwin2201/hello-react-rails
```
Navigate to the folder
```
cd hello-react-rails
```
Checkout the branch
```
git checkout setup-branch
```
Screenshot (2)

#### Thanks for reviewing my PR :blush:

### CODE REVIEW

File: app/javascript/components/message.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import { useSelector } from "react-redux";
+
+const Message = () => {
+  const greeting = useSelector((state) => state.message);
+     return ( 
+          <h1>{greeting.message}</h1>
+      );
+}
+ 
+export default Message;
```
Comment (mckent05) [inline app/javascript/components/message.js]: - Good job on the job done so far, but the frontend of your project does not seem to be rendering, Please fix this 🙏.
Your project:
ethaethaeh

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @soesandarwin2201, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Your backend server works fine, well-done 👏 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/components/message.js (hunk 1 above)
Comment (soesandarwin2201) [reply app/javascript/components/message.js]: @mckent05  i was trying to do with the ./bin/dev but i didnot past the database , can you help me with that ? please

File: app/controllers/api/messages_controller.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 class Api::MessagesController < Api::ApplicationController
   def index
     @messages = Message.find_by(id: rand(1..Message.count))
-    response.headers['Access-Control-Allow-Origin'] = 'http://127.0.0.1:3000'
-    response.headers['Access-Control-Allow-Methods'] = 'GET'
-    response.headers['Access-Control-Allow-Headers'] = 'Content-Type'
+    # response.headers['Access-Control-Allow-Origin'] = 'http://127.0.0.1:3000'
+    # response.headers['Access-Control-Allow-Methods'] = 'GET'
+    # response.headers['Access-Control-Allow-Headers'] = 'Content-Type'
```
Comment (HAMMAS-SALEEM) [inline app/controllers/api/messages_controller.rb]: - [ ] Great work on making the requested change but i guess while making them you left some commented piece of code. Kindly remove the commented code to make your codebase clean and to follow Ruby best practices :+1:

File: app/javascript/redux/Slice.js [hunk 3]
```diff
@@ -10,6 +10,7 @@ const initialState = {
 export const getMessageData = createAsyncThunk('message/getMessageData', async () => {
      const response = await fetch(API);
      const data = await response.json();
+     console.log(data);
      return data;
```
Comment (HAMMAS-SALEEM) [inline app/javascript/redux/Slice.js]: **[OPTIONAL BUT HIGHLY RECOMMENDED]**

- [ ] Kindly remove the console log to follow the JavaScript best practices :+1: and also it kills the performance of your application :fearful:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @soesandarwin2201 :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requested changes, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @soesandarwin2201, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Git Flow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**